Source/Core/Core/HW/DSPHLE/UCodes/AX.cpp

```diff
@@ -291,42 +291,33 @@ AXMixControl AXUCode::ConvertMixerControl(u32 mixer_control)
   // TODO: find other UCode versions with different mixer_control values
   if (m_crc == 0x4e8a8b21)
   {
-    ret |= MIX_L | MIX_R;
+    ret |= MIX_MAIN_L | MIX_MAIN_R;
     if (mixer_control & 0x0001)
       ret |= MIX_AUXA_L | MIX_AUXA_R;
     if (mixer_control & 0x0002)
       ret |= MIX_AUXB_L | MIX_AUXB_R;
     if (mixer_control & 0x0004)
     {
-      ret |= MIX_S;
+      ret |= MIX_MAIN_S;
       if (ret & MIX_AUXA_L)
         ret |= MIX_AUXA_S;
       if (ret & MIX_AUXB_L)
         ret |= MIX_AUXB_S;
     }
     if (mixer_control & 0x0008)
-    {
-      ret |= MIX_L_RAMP | MIX_R_RAMP;
-      if (ret & MIX_AUXA_L)
-        ret |= MIX_AUXA_L_RAMP | MIX_AUXA_R_RAMP;
-      if (ret & MIX_AUXB_L)
-        ret |= MIX_AUXB_L_RAMP | MIX_AUXB_R_RAMP;
-      if (ret & MIX_AUXA_S)
-        ret |= MIX_AUXA_S_RAMP;
-      if (ret & MIX_AUXB_S)
-        ret |= MIX_AUXB_S_RAMP;
-    }
+      ret |= MIX_ALL_RAMPS;
```

Comment:
Am I correct in guessing that the intended change here is equivalent to adding `| MIX_MAIN_S_RAMP` to the first line, like it's done below if m_crc is not 0x4e8a8b21?  

If so this consolidation seems wrong to me, but I'm not familiar with the DSP so I could be missing context. For example, if the 0x8 bit of mixer_control is set and the lowest three bits are not then in the old code none of these ifs are true leaving ret == 0xf, while in this version it becomes 0xAAAAAF.

Reply:
1. Make sure to look at the two commits individually.
2. Yes, setting all ramp flags changes the value of `ret`. However, a ramp flag has no effect if its corresponding channel is not active. See the `MIX_ON` checks in AXVoice.h that precede each use of `RAMP_ON`.